src/common/transport/transport-client.js

```diff
@@ -0,0 +1,80 @@
+import { TransportClientError } from '../errors';
+import rp from 'request-promise';
```

Comment:
I think having a separate transporter opens the door to the Universal JS idea, what do you think of that as a goal?